sr_hand_health_report/src/sr_hand_health_report/overrun_check.py

```diff
@@ -26,58 +26,101 @@ class OverrunCheck(SrHealthReportCheck):
     PASSED_THRESHOLDS = {'overrun_average': 1, 'drop_average': 1}
 
     def __init__(self, hand_side, fingers_to_test):
+        """
+            Initialize the OverrunCheck object
+            @param hand_side: String indicating the side
+            @param fingers_to_test: List of finger prefixes to test
+        """
         super().__init__(hand_side, fingers_to_test)
 
+        self._name = "Overrun"
         self.number_of_drops = 0
         self.iterations = 0
         self.overrun_average = 0
         self.drop_average = 0
-        self._result = None
-
-        rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self.overruns_callback)
-        rospy.Subscriber(f"/{self._hand_prefix}/debug_etherCAT_data", EthercatDebug, self.drops_callback)
-
-    def overruns_callback(self, data):
-        overrun = OverrunCheck.get_recent_overruns(data)
-
-        self.overrun_average += int(float(overrun))
+        self._result = {'overrun': {}}
+
+        rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self._overruns_callback)
+        rospy.Subscriber(f"/{self._hand_prefix}/debug_etherCAT_data", EthercatDebug, self._drops_callback)
+
+    def _overruns_callback(self, data):
+        """
+            Overrun callback
+            @param: DiagnosticArray data
+        """
+        overrun_value = OverrunCheck.get_recent_overruns(data)
+        self.overrun_average += int(float(overrun_value))
```

Comment:
Byt the look of it overrun_value is a string that contains a float? :thinking: I would have thought that the number of overruns is discrete and not continuous. In addition, why doesn't your getter return an int directly?

Reply:
Oh, I didn't notice this before (copied from a different part in our codebase). Changed to return a casted int from the get_recent_overruns, makes sense?

Reply:
Do you mean that `get_recent_overruns()` now returns an int?